scrapers/torrent.py

```diff
@@ -1,66 +1,139 @@
+        if b"taglist" in torrent_data[b"metadata"]:
+            res["performers"]=[{"name":x} for x in process_tags_performers(torrent_data[b"metadata"][b"taglist"])]
+        if b"comment" in torrent_data:
+            res["url"] = decode_bytes(torrent_data[b"comment"])
+        if b"creation date" in torrent_data:
+            res["date"] = datetime.fromtimestamp(torrent_data[b"creation date"]).strftime("%Y-%m-%d")
+    return res
+
+
+def decode_bytes(s, encodings=("utf-8", "latin-1")):
+    for enc in encodings:
+        try:
+            return s.decode(enc)
+        except UnicodeDecodeError:
+            pass
+    return s.decode("utf-8", "ignore")
+
+
+def scene_in_torrent(scene_data, torrent_data):
+    for scene in scene_data["files"]:
+        if b"length" in torrent_data[b"info"]:
+            if scene["filename"] in decode_bytes(torrent_data[b"info"][b"name"]) and torrent_data[b"info"][b"length"] == scene["size"]:
+                return True
+        elif b"files" in torrent_data[b"info"]:
+            for file in torrent_data[b"info"][b"files"]:
+                if scene["filename"] in decode_bytes(file[b"path"][-1]) and file[b"length"] == scene["size"]:
+                    return True
+
+
+def process_torrents(scene_data):
+    if scene_data:
+        for name in TORRENTS_PATH.glob("*.torrent"):
+            with open(name, "rb") as f:
+                torrent_data = bdecode(f.read())
+                if scene_in_torrent(scene_data, torrent_data):
+                    return get_torrent_metadata(torrent_data)
+    return {}
+
+def similarity_file_name(search, fileName):
+    result = difflib.SequenceMatcher(a=search.lower(), b=fileName.lower())
+    return result.ratio()
+
+def cleanup_name(name):
+    ret = str(name)
+    ret = ret.removeprefix("torrents\\").removesuffix(".torrent")
+    return ret
+
+
 if sys.argv[1] == "query":
     fragment = json.loads(sys.stdin.read())
-    title=fragment['title']
-    if '.torrent' in title:
-        lookup_torrent(title)
-    else:
-        query(title)
-    print(json.dumps(fragment))
-# Last Updated February 18, 2021
+    print(json.dumps(process_torrents(get_scene_data(fragment))))
+elif sys.argv[1] == "fragment":
+    filename = json.loads(sys.stdin.read()).get('url')
+    with open(filename, 'rb') as f:
+        torrent_data = bdecode(f.read())
+        print(json.dumps(get_torrent_metadata(torrent_data)))
+elif sys.argv[1] == "search":
+    search = json.loads(sys.stdin.read()).get('name')
+    torrents = list(TORRENTS_PATH.rglob('*.torrent'))
+    ratios = {}
+    for t in torrents:
+        clean_t = cleanup_name(t)
+        ratios[round(10000*(1-similarity_file_name(search, clean_t)))] = {'url': str(t.absolute()), 'title': clean_t}
+
+    # Order ratios
+    ratios_sorted = dict(sorted(ratios.items()))
+    # Only return the top 5 results
+    if len(ratios) > 5:
+        ratios = ratios_sorted[5:]
```

Comment:
Small error here, can't fix it myself anymore. This and the next line should be replaced with:
`if len(ratios) > 5:
        ratios_sorted = list(ratios_sorted.values())[0:5]
    else:
        ratios_sorted = ratios_sorted.values()
    
    print(json.dumps(list(ratios_sorted)))`